### PR TITLE
Fix inconsistencies with the CLI `-h` flag

### PR DESCRIPTION
**Pull request by maciekgrzybek**
Closes: #1120

### What are the changes and their implications?
- update the pre-run hook to not to error out valid commands
- add flags to the commands

### CODE REVIEW

**Comment by flybayer** [comment]
Thanks @maciekgrzybek! Looks like you have a failing test. Once that's fixed, we can merge :)

**Comment by flybayer** [comment]
@maciekgrzybek are you able to fix the failing test? Also have a a merge conflict

**Comment by maciekgrzybek** [comment]
@flybayer apologies, I'll try to fix this today :)

**Comment by maciekgrzybek** [comment]
@flybayer I'm actually struggling to understand what's causing the test to fail here. I've asked for help on slack so maybe someone will help me out :)

File: packages/cli/src/commands/build.ts [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
+
   async run() {
     const config = {
       rootFolder: process.cwd(),
     }
 
     try {
+      // @ts-ignore noUnusedLocals
+      // This is needed to make sure that help flag is working correctly
+      const {flags}: {flags: Flags} = this.parse(Build)
```
Comment (flybayer) [inline packages/cli/src/commands/build.ts:22]: You can clean this up to be just `this.parse(Build)`. And you can move it up outside the try catch.

File: packages/cli/src/commands/build.ts [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
+
       const build: typeof ServerBuild = require("@blitzjs/server").build
       await build(config)
     } catch (err) {
-      console.error(err)
-      process.exit(1) // clean up?
+      if (err instanceof PromptAbortedError) this.exit(0)
+      this.error(err)
```
Comment (flybayer) [inline packages/cli/src/commands/build.ts:28]: Can you undo this change?

File: packages/cli/src/commands/console.ts [hunk 3]
```diff
@@ -12,7 +12,15 @@ export class Console extends Command {
...
   }
 
+  static flags = {
+    help: flags.help({char: "h"}),
+  }
+
   async run() {
+    // @ts-ignore noUnusedLocals
+    // This is needed to make sure that help flag is working correctly
+    const {flags} = this.parse(Console)
```
Comment (flybayer) [inline packages/cli/src/commands/console.ts:22]: Again, all you need is `this.parse(Console)`

File: packages/cli/src/commands/install.ts [hunk 4]
```diff
@@ -144,6 +150,10 @@ export class Install extends Command {
   }
 
   async run() {
+    // @ts-ignore noUnusedLocals
+    // This is needed to make sure that help flag is working correctly
+    const {flags}: {flags: Flags} = this.parse(Install)
```
Comment (flybayer) [inline packages/cli/src/commands/install.ts:155]: Same thing

File: packages/cli/src/commands/test.ts [hunk 5]
```diff
@@ -12,8 +12,15 @@ export class Test extends Command {
...
 
+  static flags = {
+    help: flags.help({char: "h"}),
+  }
+
   async run() {
-    const {args} = this.parse(Test)
+    // @ts-ignore noUnusedLocals
+    // This is needed to make sure that help flag is working correctly
+    const {args, flags} = this.parse(Test)
```
Comment (flybayer) [inline packages/cli/src/commands/test.ts:22]: same, leave out `flags`

**Review by flybayer — COMMENTED** [review]
Ok, finally got to this! I fixed the test for you. The test was using .prototype which wasn't correct (something someone else did previously).

Also have some other changes for you to make

File: packages/cli/src/commands/install.ts (hunk 4 above)
Comment (maciekgrzybek) [reply packages/cli/src/commands/install.ts]: Done 👍

File: packages/cli/src/commands/console.ts (hunk 3 above)
Comment (maciekgrzybek) [reply packages/cli/src/commands/console.ts]: Done 👍

File: packages/cli/src/commands/build.ts (hunk 2 above)
Comment (maciekgrzybek) [reply packages/cli/src/commands/build.ts]: Done 👍

File: packages/cli/src/commands/build.ts (hunk 1 above)
Comment (maciekgrzybek) [reply packages/cli/src/commands/build.ts]: Done 👍

**Comment by maciekgrzybek** [comment]
Amazing, thanks for help @flybayer :) weird thing is, I've tried the `prototype` fix and it didn't work for me... 🤔  Maybe I missed something.. :) Anyway, I made the changes, hopefully, it's ok now :)

File: .vscode/settings.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+  "workbench.colorCustomizations": {
+    "activityBar.background": "#1A2548",
+    "titleBar.activeBackground": "#243465",
+    "titleBar.activeForeground": "#F8F9FD"
+  }
+}
```
Comment (flybayer) [inline .vscode/settings.json:7]: Can you remove this file? Probably makes sense to add `.vscode` to the root `.gitignore`

File: packages/cli/src/commands/test.ts [hunk 7]
```diff
@@ -12,8 +12,15 @@ export class Test extends Command {
     },
   ]
 
+  static flags = {
+    help: flags.help({char: "h"}),
+  }
+
   async run() {
+    // @ts-ignore noUnusedLocals
+    // This is needed to make sure that help flag is working correctly
```
Comment (flybayer) [inline packages/cli/src/commands/test.ts:21]:
```suggestion
```

**Review by flybayer — COMMENTED** [review]
Well weird, the test is still failing here on CI, but passes locally. I just pushed another change to maybe fix it.

Also two more things from your side:

File: .vscode/settings.json (hunk 6 above)
Comment (maciekgrzybek) [reply .vscode/settings.json:7]: Yeah missed that.. sorry removing 🆗

**Comment by maciekgrzybek** [comment]
I've accidentally removed your commit... could you please push it up again? Sorry and thanks :)

**Comment by flybayer** [comment]
Woohoo! Thanks for your help @maciekgrzybek 

@all-contributors add @maciekgrzybek for code

**Comment by maciekgrzybek** [comment]
Happy to help :)
